tensorflow/lite/experimental/micro/examples/micro_recognition/disco/image_provider.cc

```diff
@@ -0,0 +1,33 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/lite/experimental/micro/examples/micro_recognition/image_provider.h"
+
+#include "BSP_DISCO_F746NG/Drivers/BSP/STM32746G-Discovery/stm32746g_discovery_camera.h"
+
+TfLiteStatus InitCamera(tflite::ErrorReporter* error_reporter) {
+  if (BSP_CAMERA_Init(RESOLUTION_R160x120) != CAMERA_OK) {
+    error_reporter->Report("Failed to init camera.\n");
+    return kTfLiteError;
+  }
+
+  return kTfLiteOk;
+}
+
+TfLiteStatus GetImage(tflite::ErrorReporter* error_reporter, int frame_width,
+                      int frame_height, int channels, uint8_t* frame) {
+  BSP_CAMERA_SnapshotStart(frame);
```

Comment:
only the parameter `frame` is used here and why do you need to declare the other 3 parameters?

Reply:
I used the GetImage() function from the micro speech example as a template for this function, since I'd thought it would be good to be consistent across the examples. But I'll (void) the unused parameters to avoid compilation warning, and I'll add a comment explaining why some parameters are unused.